### PR TITLE
ZEPPELIN-1199. Need to login using keytab and principal before creating SparkContext in secured cluster

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?
Need to login using keytab and principal before creating SparkContext in secured cluster. 

### What type of PR is it?
[Bug Fix]

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-1199

### How should this be tested?
Manually verified in a secured cluster. 

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

### CODE REVIEW

**Comment by zjffdu** [comment]
Might need to check whether it works in pyspark/sparkr.

**Comment by zjffdu** [comment]
pyspark and sprakr both create SparkContext, so this fix also works in pyspark & sparkr

File: spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java [hunk 1]
```diff
@@ -337,6 +339,15 @@ public SparkContext createSparkContext() {
       conf.set("spark.yarn.isPython", "true");
     }
 
+    if (conf.contains("spark.yarn.keytab") && conf.contains("spark.yarn.principal")) {
+      try {
+        String keytab = conf.get("spark.yarn.keytab");
```
Comment (bikassaha) [inline spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java]: Why are we logging on with spark keytab instead of zeppelin keytab?

Comment (zjffdu) [inline spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java]: This is not zeppelin keytab. It is just keytab specified by zeppelin. Every interpreter has its own setting, and this is just one of the setting.

**Comment by prabhjyotsingh** [comment]
@zjffdu can you resolve conflict ?

**Comment by zjffdu** [comment]
@prabhjyotsingh Conflicts resolved.

**Comment by prabhjyotsingh** [comment]
LGTM

**Comment by zjffdu** [comment]
Close it and merge it to #1237

**Pull request closed without merging**